modules/certificate/certificates.permission.js

```diff
@@ -1,18 +1,40 @@
-const { hasRoles } = require('../../utils/rules');
+const { and, allow } = require('graphql-shield');
+const {
+  hasRoles,
+  isAuthorized,
+  isUnlocked,
+  inputDataValidation,
+} = require('../../utils/rules');
+const {
+  certificateNameValidator,
+} = require('../../validators/certificate.validator');
+
 const {
   roles: { USER, ADMIN, SUPERADMIN },
 } = require('../../consts');
 
 const certificatePermissionsQuery = {
-  getCertificateById: hasRoles([USER, ADMIN, SUPERADMIN]),
-  getAllCertificates: hasRoles([USER, ADMIN, SUPERADMIN]),
+  getCertificateById: hasRoles([ADMIN, SUPERADMIN]),
+
+  getAllCertificates: and(isAuthorized, isUnlocked),
 };
 
 const certificatePermissionsMutations = {
-  updateCertificate: hasRoles([USER, ADMIN, SUPERADMIN]),
-  addCertificate: hasRoles([USER, ADMIN, SUPERADMIN]),
+  updateCertificate: and(
+    allow,
+    inputDataValidation('name', certificateNameValidator)
```

Comment:
why name? whole input from ui called a name?

Reply:
Yes, maybe we should rename it to code. But don't think it will dramatically change the situation